File: BubbleSort.py
def bubble_sort(my_list): #function takes in a list 
    for i in range(len(my_list) -1, 0 ,-1): #reduces length of list every loop since the last number is 'sorted'
        for j in range(i): #loop moves forward through the list
            if my_list[j] > my_list[j+1]: #if current number greater than the next,swap them.
                temp = my_list[j]
                my_list[j] = my_list[j+1]
                my_list[j+1] = temp
    return my_list #returns sorted list

print(bubble_sort([4,2,6,5,1,3])) #sample input to test function            

File: c_implementation/bubble_sort.c
#include <stdio.h>
#include <time.h>

void bubble_sort(int my_list[], int n) {
    int i, j;
    // outer loop for passes
    for (i = n - 1; i > 0; i--) {
        // inner loop for comparisons
        for (j = 0; j < i; j++) {
            // if current element is greater than the next then swap the elements
            if (my_list[j] > my_list[j + 1]) {
                int temp = my_list[j];
                my_list[j] = my_list[j + 1];
                my_list[j + 1] = temp;
            }
        }
    }
}

int main() {
    int my_list[] = {7, 3, 6, 9, 4, 1};
    // calculate the size of the array
    int n = sizeof(my_list) / sizeof(my_list[0]);
    
    // Perform bubble sort with timer
    clock_t start = clock();
    bubble_sort(my_list, n);
    clock_t end = clock();
    double time_taken = ((double) (end - start)) * 1000.0 / CLOCKS_PER_SEC;

    // Print the sorted list
    printf("Sorted list: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", my_list[i]);
    }
    printf("\n");
    printf("Bubble sort completed in: %f ms\n", time_taken);

    return 0;
}
